WordPress/src/main/java/org/wordpress/android/ui/sitecreation/creation/FetchWpComSiteUseCase.kt

```diff
@@ -0,0 +1,60 @@
+package org.wordpress.android.ui.sitecreation.creation
+
+import kotlinx.coroutines.experimental.delay
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.SiteActionBuilder
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.fluxc.store.SiteStore.OnSiteChanged
+import javax.inject.Inject
+import kotlin.coroutines.experimental.Continuation
+import kotlin.coroutines.experimental.suspendCoroutine
+
+private const val FETCH_SITE_BASE_RETRY_DELAY_IN_MILLIS = 1000
+
+/**
+ * Transforms FETCH_SITE -> UPDATE_SITE fluxC request-response pair to a coroutine.
+ */
+class FetchWpComSiteUseCase @Inject constructor(
+    private val dispatcher: Dispatcher,
+    @Suppress("unused") private val siteStore: SiteStore
+) {
+    private var continuation: Continuation<OnSiteChanged>? = null
+
+    suspend fun fetchSite(remoteSiteId: Long, numberOfRetries: Int): OnSiteChanged {
+        repeat(numberOfRetries) { attemptNumber ->
+            val onSiteFetched = fetchSite(remoteSiteId)
+            if (!onSiteFetched.isError) {
+                // return only when the request succeeds
+                return onSiteFetched
+            }
+            // linear backoff
+            delay((attemptNumber + 1) * FETCH_SITE_BASE_RETRY_DELAY_IN_MILLIS) // +1 -> starts from 0
+        }
+        // return the last attempt no matter the result (success/error)
+        return fetchSite(remoteSiteId)
+    }
+
+    private suspend fun fetchSite(siteId: Long): OnSiteChanged {
+        return suspendCoroutine { cont ->
+            continuation = cont
+            dispatcher.dispatch(SiteActionBuilder.newFetchSiteAction(createSiteModel(siteId)))
+        }
+    }
+
+    private fun createSiteModel(siteId: Long): SiteModel {
```

Comment:
I think after the change in a2a7c5625995e653f781ea49553d0ae7e897321b, the original `createWpComSiteModel` is a better name here. Sorry if this wasn't clear :(

Reply:
I would have sweared I renamed it back... interesting

Fixed in [5b606c7](https://github.com/wordpress-mobile/WordPress-Android/pull/8825/commits/5b606c73b72526192808d5d0c9e0ba3cb174edba)